ProjFS.Linux/PrjFSLib.Linux.Managed/VirtualizationInstance.cs

```diff
@@ -26,8 +26,10 @@ public static Result ConvertDirectoryToVirtualizationRoot(string fullPath)
         }
 
         public virtual Result StartVirtualizationInstance(
+            string storageRootFullPath,
             string virtualizationRootFullPath,
-            uint poolThreadCount)
+            uint poolThreadCount,
+            out IntPtr mountHandle)
```

Comment:
It doesn't seem like there's a need to expose the `mountHandle` in this managed API. The `VirtualizationInstance` could just track it as a private member variable.

That said, I'm not really clear on why you need this handle in the first place. Can you give us more detail on how it's being used in the C lib? Can it just be contained to that code and not exposed out of the API at all?

Reply:
I can't see an obvious way to keep this internal.  In the MacOS implementation, you can start a background event loop to listen for kernel events via Grand Central Dispatch, and that just gets shut down for you later by the OS.  But in the Linux world, we need a way to keep a handle to a struct which records the event loop thread ID, so we can stop it later.

Reply:
To address your second point first - how will you handle a process crash or kill? You can't guarantee that your stop function will be called, so do we need some other way to clean up in that situation?

As for the handle, I agree that you need one. My assertion is that you can probably keep it contained within the C lib, and not bubble it up out of your API. The API's already exposes a "handle" - the virtualization root path. You can map that to whatever internal handle you like, and not leak implementation details out of the API. Is there anything that would prevent that approach?

Reply:
On a crash or kill, there's not much that can be done; we can't trap `SIGKILL`, obviously.  The user would have to `umount` manually to get back to a clean state by hand.  However, on startup we can try to detect this (an existing mount) and either warn or attempt to clean up automatically, then remount.

On `SIGTERM` things are a little different; it's not ideal to have a library set up its own handler (as signal handlers are global across the whole process) so typically the application does that.  But, I could put a `SIGTERM` handler into the "translation" layer (see below).

Re the handle, I see where you're coming from now.  It's non-idiomatic in standard C library terms, but we could have the VFSforGit "translation" layer keep a global path-to-struct map and lookup the root path on the stop function.  So if you prefer, I can move that into "translation" C code (all the VFSforGit API function names and wrappers) which the library only builds if you choose the build-for-VFSforGit configuration option.  Otherwise you'd get only the more conventional C approach.  Since it sounds like that would likely be your strong preference, let me take a stab at that.

Reply:
I'd rather you not have a "build-for-VFSForGit" flag :-). That path-to-handle translation could also be done in the C# interop layer, just to avoid exposing it to the C# provider. And actually, you wouldn't need to do any real mapping - just have `VirtualizationInstance` store the handle, and not bubble it up any higher than that. From the C# provider's perspective, the `VirtualizationInstance` object _is_ the handle.

Reply:
Yes, I like that better, thank you -- and it avoids the problem that if we try to retain state, in the form of paths to mount points, in the library, then it becomes a problem if the user renames some directory above the mount point (e.g., from /repos/this-gvfs-repo to /my-repos/this-gvfs-repo).  I can definitely just keep the handle in `PrjFS.Linux.VirtualizationInstace`.  Thanks!!

Reply:
I think this is fully implemented now, as described.  Thanks for the critique, that was helpful!